### ISSUE TITLE
Restore creates dublicates 

### ISSUE THREAD

**Issue by Solar-Simon** [tmo1/sms-ie#91]
Hello there, for some time I switched several times between two phones. Sms-ie is the best solution for me to quickly move my Sms and call logs.

After some time I realized Sms counter has exploded. Now I have several duplicate of my Sms.

So recovery of the backup is more a inclusion of the backup. Guess I always had to erase existing Messages first.

My expected behavior would be, when I restore a backup, I get exactly the data of the backup. And not a mix of existing data and backup. 

Now that I now, it's easy to change my workflow. But how do I remove all that Duplicates?

**Comment by tmo1** [comment]
The behavior you describe, as well as the solution you arrived at of wiping messages before restoring, [is discussed in the README](https://github.com/tmo1/sms-ie#limitations). The reason the app does not attempt to avoid message duplication is that it can be tricky to ascertain if two similar looking messages are identical, when some of the metadata is the same but some different.

I don't currently have a good way to get rid of duplicate messages in an existing collection of messages. It would not be too difficult to write a script to identify and remove duplicates, once the above problem has been addressed.

**Comment by tmo1** [comment]
I'm going to leave this issue open, as a place for discussion of the feasibility and desirability of implementing some sort of deduplication, but I don't currently have any plans to implement anything.

**Comment by maverick74** [comment]
If not a duplicate, this is, at least, related to #58 .

@tmo1 That SMS-DB system you talked about to avoid duplicates would be indeed very handy...

**Comment by tmo1** [comment]
Noted. I'm not sure I plan to implement deduplication in SMS I/E, but I'll consider it.

**Comment by tmo1** [comment]
Not at present - sorry. If I do implement something, I'll try to remember to post it here.

**Comment by farfromrefug** [comment]
@tmo1 could you implement a wipe for call.logs ? Right now I end up with duplicates after backup/restore

**Comment by tmo1** [comment]
> could you implement a wipe for call.logs ?

[Android has call log wiping built in](https://support.google.com/phoneapp/answer/2811854?hl=en#zippy=%2Cdelete-your-entire-call-history).

**Comment by farfromrefug** [comment]
@tmo1 I see. It can be done programmatically as other apps do it. Would be nice to have it in the app. Maybe even as a prompt option when importing calls.

**Comment by tmo1** [comment]
> It can be done programmatically as other apps do it. Would be nice to have it in the app. Maybe even as a prompt option when importing calls.

I'm not sure I see the point in duplicating native Android functionality, but I'll consider it.

**Comment by farfromrefug** [comment]
@tmo1 It is all about user experience. like the fact that you dont propose to set the dedaur message app from the app.
it is like permissions. they can be set from app info but as app devs we ask them from our apps to have a better user experience.

**Comment by mribbons** [comment]
This isn't easy to implement on device.

I vibed up a python tool to do it on a laptop etc.

https://github.com/mribbons/sms-ie/tree/merge-exports/MergeExports

Happy switching.

**Comment by tmo1** [comment]
> This isn't easy to implement on device.

I agree that it (like most things) is easier to do on a normal computer, but SMS I/E actually has had [message](https://github.com/tmo1/sms-ie/commit/9036005dc3db4a7c21764f93f773240f85b12ac9) and [call log](https://github.com/punpunsx/sms-ie/commit/69d2f95b9b1f0eb41168330707cf9f47f2004e5e) deduplication for years already.

> I vibed up a python tool to do it on a laptop etc.

Neat, thanks!

**Comment by Suncatcher** [comment]
> SMS I/E actually has had [message](https://github.com/tmo1/sms-ie/commit/9036005dc3db4a7c21764f93f773240f85b12ac9) and [call log](https://github.com/punpunsx/sms-ie/commit/69d2f95b9b1f0eb41168330707cf9f47f2004e5e) deduplication for years already.

does that mean all SMS and calls are checked for duplicates during restore? what about contacts?

**Comment by tmo1** [comment]
> does that mean all SMS and calls are checked for duplicated during restore?

Yes, if the deduplication setting is enabled.

> what about contacts?

Contact deduplication is not currently implemented.
